{{cookiecutter.project_slug}}/src/containers/DefaultLayout/DefaultLayout.js

```diff
@@ -0,0 +1,137 @@
+import React, { Component, Suspense } from 'react';
+import { Redirect, Route, Switch } from 'react-router-dom';
+import * as router from 'react-router-dom';
+import { Container } from 'reactstrap';
+import { connect } from 'react-redux';
+import {
+  AppAside,
+  AppFooter,
+  AppHeader,
+  AppSidebar,
+  AppSidebarFooter,
+  AppSidebarForm,
+  AppSidebarHeader,
+  AppSidebarMinimizer,
+  AppBreadcrumb2 as AppBreadcrumb,
+  AppSidebarNav2 as AppSidebarNav,
+} from '@coreui/react';
+
+// sidebar nav config
+import {loggedRoutes, publicRoutes} from '../../_nav';
+// routes config
+import routes from '../../routes';
+import * as emailAuthActions from '../../features/EmailAuth/redux/actions';
+
+const DefaultAside = React.lazy(() => import('./DefaultAside'));
+const DefaultFooter = React.lazy(() => import('./DefaultFooter'));
+const DefaultHeader = React.lazy(() => import('./DefaultHeader'));
+
+class DefaultLayout extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      navItems: publicRoutes
+    };
+  }
+
+  authCheck() {
+    if (!this.props.loggedIn) {
+      this.setState({
+        navItems: publicRoutes
+      })
+      this.props.history.push(`/login`);
+    } else {
+      this.setState({
+        navItems: loggedRoutes
+      })
+    }
+  }
+
+  componentDidMount() {
+    this.authCheck()
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.props.loggedIn !== prevProps.loggedIn) {
+      this.authCheck()
+    }
+  }
+
+  loading = () => <div className="animated fadeIn pt-1 text-center">Loading...</div>
+
+  signOut = async (e) => {
+    const { logout } = this.props;
+    await logout();
+    this.props.history.push(`/login`);
+  }
+
+  render() {
+
+    return (
+      <div className="app">
+        <AppHeader fixed>
+          <Suspense fallback={this.loading()}>
+            <DefaultHeader onLogout={e=>this.signOut(e)}/>
+          </Suspense>
+        </AppHeader>
+        <div className="app-body">
+          <AppSidebar fixed display="lg">
+            <AppSidebarHeader />
+            <AppSidebarForm />
+            <Suspense>
+            <AppSidebarNav navConfig={this.state.navItems} {...this.props} router={router}/>
```

Comment:
<img width="575" alt="Screenshot 2020-04-24 at 00 39 11" src="https://user-images.githubusercontent.com/7683329/80152256-04de8e80-85c4-11ea-8e2f-efc0995590c5.png">

Reply:
Pushed changes to address these issues.